src/Toggle/Toggle.js

```diff
@@ -66,6 +76,32 @@ function getStyles(props, context, state) {
     },
   };
 
+  Object.assign(styles.track,
+    trackStyle,
+    state.switched && styles.trackWhenSwitched,
+    state.switched && trackSwitchedStyle,
+    disabled && styles.trackWhenDisabled
+  );
+
+  Object.assign(styles.thumb,
+    thumbStyle,
+    state.switched && styles.thumbWhenSwitched,
+    state.switched && thumbSwitchedStyle,
+    disabled && styles.thumbWhenDisabled
+  );
+
+  if (state.switched) {
+    styles.thumb.marginLeft = 0 - styles.thumb.width;
+  }
+
+  Object.assign(styles.icon, iconStyle);
```

Comment:
As far as I have looked at the PR, that should be fine like this.
I don't have the time to have a deep look at it right now.

My previous point is: there is no longer much value at refactoring this component.
The win is limited where the risk is making a mistake along the way (especially as we have a low test coverage).
So if you have time, reducing the refactorization in this PR would help in reviewing it. Otherwise, just wait a bit

> Reapply this PR to next?

You can have a look at the `next` branch, but I'm pretty sure we do not longer have this type of issues.

@agoldis I hope we can release it with `v0.16.0` this weekend.